### PR TITLE
Extend groups-API to allow various VC types with arguments

### PR DESCRIPTION
**Pull request by przydatek**
Extend issuer's group-API: 
- add `group-types`-endpoint that returns VC types supported by the issuer
- add optional VC arguments to the group membership records

The arguments passed are stored and returned upon retrieval, but there are no checks yet whether the arguments match the VC spec of a group.  The issuer still returns `VerifiedMember`-VCs for all groups, and the switch to VCs matching the group's spec will be added in a follow-up PR.  Also, the corresponding changes to RP-API will be in a separate PR.

### CODE REVIEW

File: issuer/src/groups_api.rs [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+use std::collections::BTreeMap;
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize, Ord, PartialOrd)]
+pub enum ArgumentValue {
+    String(String),
+    Int(i32),
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize, Eq, PartialEq, Ord, PartialOrd)]
+pub struct CredentialSpec {
```
Comment (lmuntaner) [inline issuer/src/groups_api.rs:11]: Will you add a todo to import it from II?

**Review by lmuntaner — APPROVED** [review]
LGTM! Thanks!

File: issuer/src/groups_api.rs (hunk 1 above)
Comment (przydatek) [reply issuer/src/groups_api.rs:11]: Actually, I'm not sure we should use the imported types directly, as these "copies" are a part of public API, which means that any update of `vc_util` types would immediately change the public API of our issuer (hence updating dependencies may become more involved, we just had a similar situation in https://github.com/dfinity/internet-identity/pull/2476).  But I'll add more comments that these are basically copies.

Comment (lmuntaner) [reply issuer/src/groups_api.rs:11]: Thanks for the explanation!

**Pull request merged**